### DISCUSSION TITLE
unable to connect to the share album

### DISCUSSION THREAD

**Discussion by bryanzqj** [Caleb9/syno-photo-frame#13] [Q&A] [answered]
hello, not sure if anyone experienced below error when running the app? Looks like it is not able to connect to the album (have set album to be public with no password required.

This is the command that i used: syno-photo-frame {link to album}

Error message:

2024-05-04T22:54:30.524Z ERROR [syno_photo_frame] error decoding response body: expected value at line 1 column 1
Error: "Login to Synology Photos failed. Make sure the share link is pointing to a *publicly shared album*. If the album's password link protection is enabled, use the --password option with a valid password."

any help would be greatly appreciated!

**Comment by Caleb9** [comment]
Hi @bryanzqj, can you check what version of Synology Photos package is installed on your NAS?

**Reply by bryanzqj** [reply]
thanks for getting back @Caleb9! I have 1.6.2-0710 installed on my NAS

**Reply by Caleb9** [reply]
Ok, it's the same one that I have and it works here, so something else must be different. Usually this error occurs when Photos responds with a HTML page instead of JSON. Does the link work if you try to open it in a web browser (try in incognito mode so you're not logged in to DSM, same as the app on RPi would be)?

**Reply by bryanzqj** [reply]
yes, I was able to open the album in Incognito mode. The chromium browser on my pi zero 2 W is extremely slow though, not sure if that causes any issue. I have already increased the swap size (necessary for chromium browser to be installed).

**Reply by Caleb9** [reply]
Ok I meant to try it from a PC, sorry if that was not clear. But if you managed to open it on the pi itself then that's even better to know :). So the only thing that comes to my mind right now is maybe there's something special about the sharing link. Do you use something like Quick Connect? Or is it via an IP address? Do you use an alias for Photos app (the one you can set up in DSM's Control Panel / Login Portal / Applications)?

If all else fails I can try to make a custom build for you that would print out responses it gets from the API, so we can figure out what's going on. You could also build it yourself if you have a PC with Docker installed, or a bigger pi somewhere (4 or 5 I guess, zero 2 is going to struggle to compile the app), but let me know about the sharing link stuff before we go down that route.

**Reply by bryanzqj** [reply]
Being able to see the responses from API would be very helpful, thank you! I did use Quick Connect and the public link looks like: https://****.quickconnect.to/mo/sharing/******. No alias for the photo app. I installed the Pi OS instead of Pi OS lite, perhaps i should reinstalled the OS since Pi 2 really seems to struggle to launch the app.

**Reply by Caleb9** [reply, marked as answer]
Alright, I think it must be Quick Connect then. I admit I never tested the app with it, as I don't use QC myself (would be nice to implement support for it, but currently I don't have a timeline). Meanwhile, do you have a possibility to use an IP address directly? Your pi must be in the same LAN as the NAS for that to work though. If you can, try to replace the quickconnect.to part with an IP address with port number (5000 for http or 5001 for https), so e.g. http://192.168.1.42:5000/mo/sharing/** and see if that works?

**Reply by Caleb9** [reply]
Oh and yeah, pi zero 2 will struggle with a desktop environment, but the app works fine without it. You can try to either install the Lite version, or just kill the desktop (disable login to desktop for a more permanent solution).

**Reply by bryanzqj** [reply]
Awesome, will give both a try, thanks mate!

**Reply by bryanzqj** [reply]
Hey @Caleb9 , just want to let you know that reinstalling OS Lite and using the IP address instead of quick connect worked..Thanks for your help!!

**Reply by poweRalph** [reply]
> Alright, I think it must be Quick Connect then. I admit I never tested the app with it, as I don't use QC myself (would be nice to implement support for it, but currently I don't have a timeline). Meanwhile, do you have a possibility to use an IP address directly? Your pi must be in the same LAN as the NAS for that to work though. If you can, try to replace the quickconnect.to part with an IP address with port number (5000 for http or 5001 for https), so e.g. http://192.168.1.42:5000/mo/sharing/** and see if that works?

Hi, want to build a Synology Photo frame for the grandparents which live in a different location. Is there any chance to reach the Synology Server via reverse proxy? Or is the tool only working locally with its local IP address?
Thanks for your support in advance.

**Reply by Caleb9** [reply]
> Hi, want to build a Synology Photo frame for the grandparents which live in a different location. Is there any chance to reach the Synology Server via reverse proxy? Or is the tool only working locally with its local IP address? Thanks for your support in advance.

Hello. Yes, there are several ways you can go about it. You can expose the NAS to the internet (hardest to do securely), or set up a VPN connection, or (probably easiest) use TailScale or something similar. You install TailScale client on both devices and then use TailScale's IP address of the NAS to connect to it from the other device (the frame).

Here's some links:
* https://tailscale.com/docs/integrations/synology
* https://pimylifeup.com/raspberry-pi-tailscale/

HTH.

**Reply by poweRalph** [reply]
> > Hi, want to build a Synology Photo frame for the grandparents which live in a different location. Is there any chance to reach the Synology Server via reverse proxy? Or is the tool only working locally with its local IP address? Thanks for your support in advance.
> 
> Hello. Yes, there are several ways you can go about it. You can expose the NAS to the internet (hardest to do securely), or set up a VPN connection, or (probably easiest) use TailScale or something similar. You install TailScale client on both devices and then use TailScale's IP address of the NAS to connect to it from the other device (the frame).
> 
> Here's some links:
> 
> * https://tailscale.com/docs/integrations/synology
> * https://pimylifeup.com/raspberry-pi-tailscale/
> 
> HTH.

I have exposed the NAS to the Internet. My domain looks like:
http://myurl.tld/mo/sharing/CrYpt1cl3tt3rs

I still have no chance to access the album via synophotoframe. But I'm indeed able to access the folder if I navigate there manually in the browser.
This is true for both scenarios: w/ & w/o password.
If you have any hints, I'd highly appreciate your support.
Thanks in advance.

**Reply by Caleb9** [reply]
> I have exposed the NAS to the Internet. My domain looks like:
http://myurl.tld/mo/sharing/CrYpt1cl3tt3rs

So http://myurl.tld points to Synology Photos specifically, or to your NAS in general? Do you have the reverse proxy set up on the NAS for Synology Photos app (Control Panel -> Login Portal -> Applications)?

> But I'm indeed able to access the folder if I navigate there manually in the browser.

Just to be sure, you know there's a difference between a *folder* and an *album* in Synology Photos, right? You need the link to point to a shared album, not folder.

**Reply by poweRalph** [reply]
The url points directly to photos. For some reason it now works. Don't know what changed. But reverse proxy seems to work.
Thanks for your support anyhow.
